src/physics/transport.jl

```diff
@@ -222,5 +222,63 @@ function total_fluxes(dd::IMAS.dd, rho_total_fluxes::AbstractVector{<:Real}=dd.c
     return total_fluxes(dd.core_transport, dd.core_profiles.profiles_1d[], rho_total_fluxes; time0)
 end
 
+
+"""
+    calculate_diffusivities(dd; ne=:none, Te=:none, Ti=:none, omega=:none)
+
+Compute transport particle, heat, and rotation diffusivities from desired profiles.
+"""
+function calculate_diffusivities(dd::IMAS.dd; ne=:none, Te=:none, Ti=:none, ω=:none)
+
+    cs1d = IMAS.total_sources(dd)
+    cp1d = dd.core_profiles.profiles_1d[]
+    eqt = dd.equilibrium.time_slice[]
+    eqt1d = eqt.profiles_1d
+    rho_cp = cs1d.grid.rho_tor_norm
+    rho_eq = eqt1d.rho_tor_norm
+
+    # Volumes and surfaces
+    surf = IMAS.interp1d(rho_eq, eqt1d.surface).(rho_cp)
+
+    # Default profiles if not provided
+    ne = ne === :none ? cp1d.electrons.density_thermal : ne
+    Te = Te === :none ? cp1d.electrons.temperature : Te
+    Ti = Ti === :none ? cp1d.ion[1].temperature : Ti
+    ω = ω === :none ? cp1d.rotation_frequency_tor_sonic : ω
```

Comment:
add type checks and also ensure type stability

```suggestion
function calculate_diffusivities(dd::IMAS.dd{T}; ne::Vector{T}=T[], Te::Vector{T}=T[], Ti::Vector{T}=T[], ω::Vector{T}=T[]) where {T<:Real}

    cs1d = IMAS.total_sources(dd)
    cp1d = dd.core_profiles.profiles_1d[]
    eqt = dd.equilibrium.time_slice[]
    eqt1d = eqt.profiles_1d
    rho_cp = cs1d.grid.rho_tor_norm
    rho_eq = eqt1d.rho_tor_norm

    # Volumes and surfaces
    surf = IMAS.interp1d(rho_eq, eqt1d.surface).(rho_cp)

    # Default profiles if not provided
    ne = isempty(ne) ? cp1d.electrons.density_thermal : ne
    Te = isempty(Te) ? cp1d.electrons.temperature : Te
    Ti = isempty(Ti) ? cp1d.ion[1].temperature : Ti
    ω = isempty(ω) ? cp1d.rotation_frequency_tor_sonic : ω
```